playground/miden-wasm/src/utils_debug.rs

```diff
@@ -140,57 +145,42 @@ impl DebugExecutor {
     // --------------------------------------------------------------------------------------------
 
     /// print general VM state information.
-    pub fn print_vm_state(&self) -> String {
-        format!("{}", self.vm_state)
-    }
+    fn vm_state_to_output(&self) -> DebugOutput {
+        let mut asmop = AsmOpInfo::new(String::new(), 0, 0);
+        if self.vm_state.asmop.is_some() {
+            asmop = self.vm_state.asmop.clone().unwrap();
+        }
 
-    /// print all stack items.
-    pub fn print_stack(&self) -> String {
-        let mut res =
-            self.vm_state
-                .stack
-                .iter()
-                .enumerate()
-                .fold(String::new(), |mut s, (i, f)| {
-                    s.push_str(&format!("[{i}] {f}\n"));
-                    s
-                });
-        res.pop(); // removes unnecessary line-break
-        res
+        let memory: Vec<(u64, [u64; 4])> =
+            self.vm_state.memory.iter().map(|x| (x.0, self.word_to_ints(&x.1))).collect();
+
+        let output = DebugOutput {
+            clk: self.vm_state.clk,
+            op: format!{"{:?}", self.vm_state.op},
+            asmop_op: asmop.op().to_string(),
+            asmop_num_cycles: asmop.num_cycles(),
+            asmop_cycle_idx: asmop.cycle_idx(),
+            stack: self.vm_state.stack.iter().map(|x| x.as_int()).collect(),
+            memory: self.transform(memory),
+        };
+        
+        output
     }
 
-    /// print all memory entries.
-    pub fn print_memory(&self) -> String {
-        let mem_result = String::new();
-        for (address, mem) in self.vm_state.memory.iter() {
-            format!("{}{}", mem_result, Self::print_memory_data(address, mem));
-        }
-        return mem_result;
+    /// converts a word to a tuple of 4 u64s.
+    /// this is necessary because wasm_bindgen does not support arrays of arrays.
+    fn word_to_ints(&self, word: &Word) -> [u64; 4] {
+        [word[0].as_int(), word[1].as_int(), word[2].as_int(), word[3].as_int()]
```

Comment:
nit: this is not using `self`, should probably be a standalone function.

Reply:
jup